From cd5dabdd9e23ea51e870a24e81d2576ba2b2033c Mon Sep 17 00:00:00 2001
From: "Satyanarayana, Sandhya" <sandhya.satyanarayana@ti.com>
Date: Thu, 12 Jul 2012 11:56:41 +0530
Subject: [PATCH 1048/1092] ARM: OMAP: AM33XX: PM: Fix resume via abort path
 issue

This patch fixes the issue of resume not happening
correctly via abort path.

These fixes were suggested by hardware team.

Signed-off-by: Satyanarayana, Sandhya <sandhya.satyanarayana@ti.com>
---
 arch/arm/mach-omap2/sleep33xx.S | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/sleep33xx.S b/arch/arm/mach-omap2/sleep33xx.S
index 4f25904..4af2e3a 100644
--- a/arch/arm/mach-omap2/sleep33xx.S
+++ b/arch/arm/mach-omap2/sleep33xx.S
@@ -494,6 +494,21 @@ emif_self_refresh_dis:
 	str	r1, [r0]
 	str	r1, [r0, #4]
 
+/*
+ * A write to SDRAM CONFIG register triggers
+ * an init sequence and hence it must be done
+ * at the end
+ */
+	ldr r0, emif_addr_virt
+	add r0, r0, #EMIF4_0_SDRAM_CONFIG
+	ldr r4, emif_sdcfg_val
+	str r4, [r0]
+
+	mov r0, #0x2000
+wait_loop4:
+	subs   r0, r0, #1
+	bne wait_loop4
+
 	mov	r0, #7
 	ldmfd	sp!, {r4 - r11, pc}	@ restore regs and return
 
@@ -833,7 +848,7 @@ susp_io_pull_cmd2:
 resume_io_pull_data:
 	.word	0x18B
 resume_io_pull_cmd:
-	.word	0x4
+	.word	0x18B
 
 susp_vtp_ctrl_val:
 	.word	0xDEADBEEF
-- 
1.7.11.2

